### ISSUE TITLE
Container logs are overwritten when we use kubectl cluster-info dump

### ISSUE THREAD

**Issue by rameshgkwd05** [kubernetes/kubectl#931]
**What happened**:
We have a pod with init and run containers in init. When we use `kubectl cluster-info dump` command, we saw only logs of run-container and not of init-container.
The possible bug here in this for-loop: [clusterinfo_dump.go#L289](https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/clusterinfo/clusterinfo_dump.go#L289)

**What you expected to happen**:
We were expecting all containers log to be appended in a single log file. OR a separate log file per container should be created.

**How to reproduce it (as minimally and precisely as possible)**:

Use any tiny pod YAML with init container e.g.
```
apiVersion: v1
kind: Pod
metadata:
  name: tiny-pod
  namespace: default
spec:
  initContainers:
  - image: alpine
    command:
      - /bin/sh
      - "-c"
      - "echo 'Hi, I am an init container'"
    imagePullPolicy: IfNotPresent
    name: tiny-init
  containers:
  - image: alpine
    command:
      - /bin/sh
      - "-c"
      - "echo 'Hi, I am run container'"
    imagePullPolicy: IfNotPresent
    name: tiny-run
  restartPolicy: Always
```

Deploy this pod
```
$ kubectl apply -f tiny-pod.yaml
pod/tiny-pod created
```

Take cluster-info dump
```
$ mkdir -p /tmp/dump-kind
$ kubectl cluster-info dump -A --output-directory=/tmp/dump-kind
Cluster info dumped to /tmp/dump-kind
```

See logs.txt inside /tmp/dump-kind/default/tiny-pod directory

**Anything else we need to know?**:
```
$ cd /tmp/dump-kind/default/tiny-pod
$ cat logs.txt
==== START logs for container tiny-run of pod default/tiny-pod ====
Hi, I am run container
==== END logs for container tiny-run of pod default/tiny-pod ====
```

If we get logs of all containers of pod
```
$ kubectl logs tiny-pod --all-containers
Hi, I am an init container
Hi, I am run container
```

Hence our expectation was
```
$ cat logs.txt
==== START logs for container tiny-init of pod default/tiny-pod ====
Hi, I am init container
==== END logs for container tiny-run of pod default/tiny-pod ====
==== START logs for container tiny-run of pod default/tiny-pod ====
Hi, I am run container
==== END logs for container tiny-run of pod default/tiny-pod ====
```

**Environment**:
- Kubernetes client and server versions (use `kubectl version`):
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-11T18:14:22Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-12-04T07:23:47Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
```
$ kind --version
kind version 0.6.1
```
- OS (e.g: `cat /etc/os-release`):
```
$ cat /etc/os-release
NAME="SLES"
VERSION="12-SP3"
VERSION_ID="12.3"
PRETTY_NAME="SUSE Linux Enterprise Server 12 SP3"
ID="sles"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:12:sp3"
```

**Comment by SaiHarshaK** [comment]
I have just tested the issue on the master branch, the output is exactly what the author desires. The version that the author was using which gave the issue is release-1.17 which didn't have the fix yet. 
So, we can close the issue. @eddiezane

**Comment by eddiezane** [comment]
@SaiHarshaK thanks for investigating!

@rameshgkwd05 please reopen if you don't find this fixed on the latest version of kubectl.

/close

**Comment by rameshgkwd05** [comment]
HI @eddiezane I checked the previous versions e.g. 1.16
It does not have a code to collect init container logs.  See [here](https://github.com/kubernetes/kubectl/blob/release-1.16/pkg/cmd/clusterinfo/clusterinfo_dump.go#L276)

Can we backport this fix for v1.16?
